### PR TITLE
Add setup time to throughput information

### PR DESCRIPTION
**Pull request by spxiwh**
Here's a small addition to the throughput information: Specifically we now store the time it takes to make injections, condition, do gating, read data etc. This can be used to measure the fraction of time spent in setup, versus time spent in the main filtering loop. So far this is not added to the plotting code, but the information does get stored. (I have added a "mean average" entry to the plots' legends to have some number to clearly compare over runs).

This has been tested in a uberbank-style run with the injcutter stuff:

https://www.atlas.aei.uni-hannover.de/~spxiwh/LSC/aLIGO/O1/analyses/inj_cutter/chirp_time_window/o1_injcutter_v3cutter/8._workflow/8.01_throughput/

Also a test plot where I plot the new information ... Clearly the presentation is rubbish and we need to do something else. This just demonstrates that the number seems to be sane:

https://www.atlas.aei.uni-hannover.de/~spxiwh/LSC/H1L1-PLOT_THROUGHPUT_BNS01_INJ-1128299417-1083600.png

### CODE REVIEW

File: pycbc/events/events.py [hunk 1]
```diff
@@ -480,6 +481,7 @@ def __setitem__(self, name, data):
             time_ratio = numpy.array([float(self.analysis_time) / float(self.run_time)])
             temps_per_core = float(self.ntemplates) / float(self.ncores)
             f['search/templates_per_core'] = numpy.array([float(temps_per_core) * float(time_ratio)])
+            f['search/setup_time_fraction'] = numpy.array([float(self.setup_time) / float(self.run_time)])
```
Comment (ahnitz) [inline pycbc/events/events.py]: __[flake8]__
* [E501] line too long (106 > 79 characters)

Comment (spxiwh) [inline pycbc/events/events.py]: Alex is never happy!

Comment (spxiwh) [inline pycbc/events/events.py]: Fixed

**Comment by josh-willis** [comment]
So, the throughput numbers seem very high to me (from your example)---is there some other change that could explain this? You're seeing an average of 20-30 ktemplates/core for injection runs, and I wouldn't naively expect injection cutting to affect this much.  Or was this run also done where you did some injection culling inside the loop? I don't see such changes in this pull request so I had assumed not.

I think we might also want to consider putting some of this information into either a caption or elsewhere on the page, rather than dumping it all into the plot legend.  But I wasn't sure on the machinery to do that, which is why I hadn't already.

**Comment by spxiwh** [comment]
Hi Josh, Yes there is some injection cutting stuff going on in the example both inside the loop and outside (and I can get injection cutting a *lot* higher than 20 - 30 K:

https://www.atlas.aei.uni-hannover.de/~spxiwh/LSC/aLIGO/O1/analyses/inj_cutter/chirp_time_window/o1_injcutter_v3cutter_run3/8._workflow/8.01_throughput/

in that example I am currently storing as ntemplates the full number of templates in the bank, which of course is not the amount actually being filtered, it's just being used as a comparison. What exactly to store and show on these plots when doing this is still an open question. For the injection cutting testing I want to see that actual speed up, and this is a good measure of that)

*BUT* that is nothing to do with this pull-request, as there are no injection cutting related changes here. Do you have any issues with the changes contained here? Probably the injection cutting stuff needs quite a bit more discussion and testing before merging and some of that needs to be about what gets shown in throughput plots, but it made sense to pull these changes off and get these pushed.

**Comment by josh-willis** [comment]
Since those changes are not in this pull request, then no I don't have a problem merging what is here.  I think we still want to revisit/improve what we show and how, but that also can wait future pull requests.

**Pull request merged**